app/src/main/AndroidManifest.xml

```diff
@@ -17,7 +17,7 @@
         android:theme="@style/AppTheme">
         <activity
             android:name=".view.MainActivity"
-            android:launchMode="singleInstance"
+            android:launchMode="singleTask"
```

Comment:
this way MainActivity will always be the root of the stack, and navigating to it will finish all upper stack.